optuna/samplers/_cmaes.py

```diff
@@ -365,7 +401,7 @@ def _to_optuna_param(distribution: BaseDistribution, cma_param: float) -> Any:
     return cma_param
 
 
-def _initialize_x0(search_space: Dict[str, BaseDistribution]) -> Dict[str, np.ndarray]:
+def _initialize_x0(search_space: Dict[str, BaseDistribution]) -> Dict[str, Any]:
```

Comment:
If `self._x0` is set in `_init_optimizer()`, its type is guaranteed as `Dict[str, np.ndarray]`.

Reply:
We defined `self._x0` like the following.

https://github.com/optuna/optuna/blob/79fadf4b7a221bc55656d881f45ded60402bf2fd/optuna/samplers/_cmaes.py#L126-L137

And `_initialize_x0` returns external representation (e.g. it returns integer object if given IntUniformDistribution).

https://github.com/optuna/optuna/blob/79fadf4b7a221bc55656d881f45ded60402bf2fd/optuna/samplers/_cmaes.py#L404-L412

So I guess `Dict[str, Any]` is correct. What do you think?

Reply:
Hmm... `DiscreteUniformDistribution` returns internal representation and `IntLogUniformDistribution` returns float value 🤔  These values looks like internal representation.

https://github.com/optuna/optuna/blob/79fadf4b7a221bc55656d881f45ded60402bf2fd/optuna/samplers/_cmaes.py#L413-L416

Reply:
Thank you for pointing it out. I found a bug of x0 initialization and fix it. Could you check my following comments?

* https://github.com/optuna/optuna/pull/1548#discussion_r463950014
* https://github.com/optuna/optuna/pull/1548#discussion_r463950471